lib/rubygems/deprecate.rb

```diff
@@ -45,13 +45,38 @@ def self.next_rubygems_major_version # :nodoc:
     Gem::Version.new(Gem.rubygems_version.segments.first).bump
   end
 
+  ##
+  # Simple deprecation method that deprecates +name+ by wrapping it up
+  # in a dummy method. It warns on each call to the dummy method
+  # telling the user of +repl+ (unless +repl+ is :none) and the
+  # year/month that it is planned to go away.
+
+  def deprecate(name, repl, year, month)
+    class_eval do
+      old = "_deprecated_#{name}"
+      alias_method old, name
+      define_method name do |*args, &block|
+        klass = self.kind_of? Module
+        target = klass ? "#{self}." : "#{self.class}#"
+        msg = [ "NOTE: #{target}#{name} is deprecated",
+                repl == :none ? " with no replacement" : "; use #{repl} instead",
+                ". It will be removed on or after %4d-%02d-01." % [year, month],
+                "\n#{target}#{name} called from #{Gem.location_of_caller.join(":")}",
+        ]
```

Comment:
I believe we are introducing here the same kind of issue we're trying to solve in #3506: we're showing a deprecation message only meant for gem authors to all users. And we're showing this deprecation message about the deprecation system right next to the deprecation of a method that end users _do_ care about.

I feel we might cause brain damage to our users :laughing:.

So, I believe we should revert the deprecation of `Gem::Deprecate` for now until we figure out good strategies to be smart about whether we are in a end user context or in a gem development context, and apply them to this particular case.

Reply:
I have missed that. I agree on removing this deprecation warning for now.

Reply:
FWIW, I placed this comment at the wrong place, my comments refers to the line _below_ this one, just to clarify.

Reply:
I'm not sure if there's other way to do this without end users seeing the deprecation warning. 🤔 

This is not something that we could check at build time...

Reply:
I think we can try to approximate it. I first idea that came to my mind was whether there's a ".gemspec" file at the CWD or up in the directory hierarchy.

Reply:
Can we just remove it for now (should I provide PR?) and discuss this as a follow-up in new issue?

Reply:
Yes, yes, that's exactly what I was suggesting.

Reply:
https://github.com/rubygems/rubygems/pull/3536